Git-egylet/pacman.py

```diff
@@ -0,0 +1,242 @@
+        :param pos_y: horizontal coordinate
+        :return: pos_x, pos_y (int): the updated coordinates
+        """
+        pos_y += 1
+        return pos_x, pos_y
+
+    def _down(self, pos_x: int, pos_y: int):
+        """
+        This function moves Pacman downwards.
+        :param pos_x: vertical coordinate
+        :param pos_y: horizontal coordinate
+        :return: pos_x, pos_y (int): the updated coordinates
+        """
+        pos_x += 1
+        return pos_x, pos_y
+
+    def _left(self, pos_x: int, pos_y: int):
+        """
+        This function moves Pacman left
+        :param pos_x: vertical coordinate
+        :param pos_y: horizontal coordinate
+        :return: pos_x, pos_y (int): the updated coordinates
+        """
+        pos_y -= 1
+        return pos_x, pos_y
+
+    def _check_pellets(self, x: int, y: int, score: int):
+        """
+        This function checks if Pacman has interacted with a pellet.
+        If so, the score increases and the pellet is removed from the map.
+        :param x: horizontal coordinate
+        :param y: vertical coordinate
+        :return: None
+        """
+        if (x, y) in self.pellets:
+            score += 1
+            self.pellets.remove((x, y))
+        return score
+
+    def _create_observation(self):
+        """
+        This function creates a grayscale observation from the current state of the game.
+        :return: obs_ (ndarray): the created observation
+        """
+        # map initialization
+        obs_ = np.zeros((self.world_size, self.world_size, 1))
+        # add pellets
+        for pellet in self.pellets:
+            obs_[pellet[0], pellet[1], 0] = 0.25
+        # add Pacman
+        body_coords = self.body
+        obs_[body_coords[0], body_coords[1], 0] = 0.8
+        return obs_
+
+    def render(self):
+        """
+        This function is responsible for the visualization of the game.
+        It creates a cv2 plot from the current game state.
+        :return: None
+        """
+        if self.last_state is not None:
+            img = np.float32(self.last_state)
+        else:
+            img = np.float32(self._create_observation())
+        img = cv2.cvtColor(img, cv2.COLOR_GRAY2BGR)
+        # rescale the image
+        for i in range(self.show_window_size):
+            for j in range(self.show_window_size):
+                self.show_window[i][j] = img[i // self.ratio][j // self.ratio]
+        cv2.imshow("Pacman", self.show_window)
+        cv2.waitKey(50)
+
+    def reset(self):
+        """
+        This function restarts the game by restoring the initial values and creating a new session.
+        :return: obs_ (ndarray): observation of the current state
+        """
+        self.body = []
+        self.pellets = []
+        self.step_counter = 0
+        self.direction = 0
+        self.score = 0
+        self.last_state = None
+
+        self.create_pacman()
+        self.create_pellets(10)
```

Comment:
Magic number, célszerű lenne ennek egy class variable, amit a konstruktorban adtok meg.
```suggestion
        self.create_pellets(self.num_pellets)
```

Reply:
Solved